Shared/alerts/DynamicEntityLayerAlertTarget.cpp

```diff
@@ -0,0 +1,197 @@
+    {
+      // get a cleanup reference to the graphic that was stored and remove it from the lookup
+      std::unique_ptr<Graphic> graphic(m_entityGraphics[dynamicEntity->entityId()]);
+      m_entityGraphics.remove(dynamicEntity->entityId());
+
+      // signal the data has changed and rebuild the quad tree
+      rebuildQuadtree();
+      emit dataChanged();
+    }
+    info->deleteLater();
+  });
+}
+
+/*!
+  \brief Destructor.
+ */
+Dsa::DynamicEntityLayerAlertTarget::~DynamicEntityLayerAlertTarget()
+{
+}
+
+/*!
+  \brief Returns the list of \l Esri::ArcGISRuntime::Geometry which are in the \a targetArea.
+
+  \note No exact intersection tests are carried out to create this list.
+ */
+QList<Geometry> DynamicEntityLayerAlertTarget::targetGeometries(const Envelope& targetArea) const
+{
+  // if the quadtree has been built, use  it to return the set of candidate geometries
+  if (m_quadtree) { return m_quadtree->candidateIntersections(targetArea); }
+
+  // otherwise, return all of the geometry in the overlay
+  QList<Geometry> geometries;
+  for (const auto* dynamicEntity : qAsConst(m_entityGraphics))
+  {
+    if (!dynamicEntity) { continue; }
+    geometries.append(dynamicEntity->geometry());
+  }
+
+  return geometries;
+}
+
+/*!
+  \brief Returns an empty QVariant.
+ */
+QVariant DynamicEntityLayerAlertTarget::targetValue() const
+{
+  return QVariant{};
+}
+
+/*!
+  \internal
+
+  Connect signals etc. for a new \a dynamicEntity.
+ */
+void DynamicEntityLayerAlertTarget::connectEntityGraphic(DynamicEntity* dynamicEntity)
+{
+  // check for an existing graphic
+  Graphic* graphic;
+  if (m_entityGraphics.contains(dynamicEntity->entityId()))
+  {
+    // update the geometry
+    graphic = m_entityGraphics[dynamicEntity->entityId()];
+    graphic->setGeometry(dynamicEntity->geometry());
+  }
+  else
+  {
+      // if no graphic existed for the entity id, construct a new one and insert it into the graphics lookup
+    graphic = new Graphic(dynamicEntity->geometry(), this);
+    connect(graphic, &Graphic::geometryChanged, this, &DynamicEntityLayerAlertTarget::dataChanged); // trigger the dataChanged signal on geometry updates
+    m_entityGraphics[dynamicEntity->entityId()] = graphic;
+
+    // if the quadtree has already been initialized, append the new graphic
+    if (m_quadtree) { m_quadtree->appendGeoElment(graphic); }
+    else
+    {
+      // otherwise call the rebuild method to initialize it
+      rebuildQuadtree();
+    }
+    emit dataChanged();
+  }
+}
+
+
+/*!
+  \brief internal.
+
+  Build the quadtree used to find intersections with entity geometry etc.
+ */
+void DynamicEntityLayerAlertTarget::rebuildQuadtree()
+{
+  // destroy the quadtree if it already existed
+  if (m_quadtree)
+  {
+    delete m_quadtree;
+    m_quadtree = nullptr;
+  }
+
+  // build a list of pointers to geoelements for every graphic in the lookup
+  QList<GeoElement*> elements;
+  for (auto* dynamicEntity : qAsConst(m_entityGraphics))
```

Comment:
ditto std::as_const (see previous comments... `qAsConst` has been deprecated in newer versions of Qt).

Reply:
Fixed.